autoPyTorch/pipeline/components/setup/traditional_ml/base_model.py

```diff
@@ -87,17 +95,18 @@ def fit(self, X: Dict[str, Any], y: Any = None) -> autoPyTorchSetupComponent:
         if 'X_test' in X.keys() and X['X_test'] is not None:
             if isinstance(X['X_test'], pd.DataFrame):
                 X['X_test'] = X['X_test'].to_numpy()
-            test_preds = self.model.predict(X_test=X['X_test'], predict_proba=True)
+            test_preds = self.model.predict(X_test=X['X_test'], predict_proba=self.model.is_classification)
             self.fit_output["test_preds"] = test_preds
         return self
 
     @abstractmethod
     def build_model(self, input_shape: Tuple[int, ...], output_shape: Tuple[int, ...],
-                    logger_port: int) -> BaseClassifier:
+                    logger_port: int, task_type: str, output_type: str) -> BaseTraditionalLearner:
         """
-        This method returns a pytorch model, that is dynamically built using
-        a self.config that is model specific, and contains the additional
-        configuration hyperparameters to build a domain specific model
+        This method returns a traditional learner, that is dynamically
+        built using a self.config that is model specific, and contains
+        the additional configuration hyperparameters to build a domain
+        specific model
```

Comment:
Can you elaborate on this? Maybe more generally like: Build model returns an estimator based on the provided configuration?